File: Src/Litium.Accelerator.Administration.Extensions/ViewModels/IndexingModel.cs
using System.Collections.Generic;

namespace Litium.Accelerator.Administration.Extensions.ViewModel
{
    public class IndexingModel
    {
        public List<Template> Templates { get; set; }

        public string Message { get; set; }

        public class Template
        {
            public string Title { get; set; }
            public string GroupingFieldId { get; set; }
            public List<FieldGroup> Fields { get; set; }
            public string TemplateId { get; set; }
        }

        public class FieldGroup
        {
            public string Title { get; set; }
            public string FieldId { get; set; }
        }
    }
}

File: Src/Litium.Accelerator.Administration.Extensions/src/AcceleratorExtensions/interfaces/indexingModel.ts
export interface IIndexingModel {
    templates: ITemplate[];
    message: string;
}

export interface ITemplate {
    title: string;
    groupingFieldId: string;
    fields: IFieldGroup[];
    templateId: string;
}

export interface IFieldGroup {
    title: string;
    fieldId: string;
}
